File: src/main/java/com/teamwizardry/librarianlib/util/DefaultedMap.java
package com.teamwizardry.librarianlib.util;

import java.util.HashMap;

public class DefaultedMap<K, V> extends HashMap<K, V> {
	private static final long serialVersionUID = 5615718886973854633L;
	
	protected V defaultValue;

	public DefaultedMap(V defaultValue) {
		this.defaultValue = defaultValue;
	}

	@Override
	public V get(Object k) {
		return this.getOrDefault(k, defaultValue);
	}
}

File: src/main/java/com/teamwizardry/librarianlib/util/DefaultedMap.kt
package com.teamwizardry.librarianlib.util

import java.util.HashMap

class DefaultedMap<K, V>(protected var defaultValue: V) : HashMap<K, V>() {

    override operator fun get(k: Any?): V {
        return (this as java.util.Map<K, V>).getOrDefault(k, defaultValue)
    }

    companion object {
        private val serialVersionUID = 5615718886973854633L
    }
}
